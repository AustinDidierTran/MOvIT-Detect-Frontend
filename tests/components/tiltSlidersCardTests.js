/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import React from 'react';
import PropTypes from 'prop-types';
import configureMockStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TiltSlidersCard from '../../src/components/tiltSlidersCard';

Enzyme.configure({ adapter: new Adapter() });

describe('TiltSlidersCard Tests', () => {
  const initialState = { applicationReducer: { language: 'en' } };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    tiltFrequecy: 40,
    tiltLength: 30,
    tiltAngle: 20,
    maxAngle: 45,
    title: 'This is a title',
    modifiable: true,
  };

  it('should have proptypes', () => {
    const actualValue = TiltSlidersCard.WrappedComponent.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      tiltFrequecy: PropTypes.number,
      tiltLength: PropTypes.number,
      tiltAngle: PropTypes.number,
      maxAngle: PropTypes.number,
      title: PropTypes.string,
      onFrequencyChange: PropTypes.func.isRequired,
      onLengthChange: PropTypes.func.isRequired,
      onAngleChange: PropTypes.func.isRequired,
      modifiable: PropTypes.bool,
      onModifie: PropTypes.func,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<TiltSlidersCard store={store} {...props} />).dive();

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
