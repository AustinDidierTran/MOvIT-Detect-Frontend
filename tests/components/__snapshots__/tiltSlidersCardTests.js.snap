// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TiltSlidersCard Tests should match the snapshot 1`] = `
<Card
  class="col-md-6"
  className={null}
  footer={null}
  header={
    <div
      className="ui-card-title"
      style={
        Object {
          "marginLeft": "14px",
          "marginTop": "14px",
        }
      }
    >
      This is a title
       Â 
      <i
        className="fa fa-check"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      />
    </div>
  }
  id={null}
  style={
    Object {
      "backgroundColor": "white",
      "boxShadow": "5px 5px 5px gainsboro",
      "padding": "5px",
    }
  }
  subtitle={null}
  title={null}
>
  <div
    className="row"
  >
    <div
      className="col-9"
    >
      <Connect(TiltSliders)
        maxAngle={45}
        tiltAngle={20}
        tiltFrequecy={40}
        tiltLength={30}
      />
    </div>
    <div
      className="col-3 justify-content-center align-self-center"
    >
      <img
        alt="chair"
        height="50"
        src={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "transform": "rotate(-20deg)",
          }
        }
        width="50"
      />
    </div>
  </div>
</Card>
`;
